runtime/tchannel_client.go

```diff
@@ -154,10 +157,13 @@ func (c *tchannelClient) Call(ctx context.Context, thriftService, methodName str
 	retryOpts := &tchannel.RetryOptions{
 		TimeoutPerAttempt: c.timeoutPerAttempt,
 	}
-	ctx, cancel := tchannel.NewContextBuilder(c.timeout).
+	ctxBuilder := tchannel.NewContextBuilder(c.timeout).
 		SetParentContext(ctx).
-		SetRetryOptions(retryOpts).
-		Build()
+		SetRetryOptions(retryOpts)
+	if c.routingKey != nil {
```

Comment:
Nit: It seems to be simpler to use `string` as type for `routingKey` and check if it is not empty string.

Reply:
+1, seems like `string` type is sufficient